test_stack.py

```diff
@@ -0,0 +1,57 @@
+# Test file for stack.py
+# Authors Mark, Efrain, Henry
+
+import pytest
+import stack as st
+
+
+def test_init():
+    """Test stack constructor."""
+    a = st.Stack()
+    assert isinstance(a, st.Stack)
+
+def test_push(empty_stack, populated_stack):
+    """Test push function, with empty and populated stacks."""
+    empty_stack.push('first')
+    assert 'first' == empty_stack.top.value
```

Comment:
might be nice to take this one step further to ensure the transition from empty to having one item doesn't hose the stack somehow.